nexus/src/external_api/params.rs

```diff
@@ -18,13 +18,25 @@ use uuid::Uuid;
 
 // Silos
 
+/// How users will be provisioned in a silo.
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+pub enum UserProvisionType {
+    /// Do not allow users to be created
```

Comment:
suggestion added in a77ccf26.

> Are we imagining it might be possible one day to connect to an IdP such that the IdP would successfully authenticate you but we would fail anyway because we didn't already have a user record for you?

This scenario would happen if there was a bug in an IdP's SCIM. They wouldn't want to set the provision type to JIT because the intention is that provisioning (and deprovisioning) is handled by SCIM calls. A user that was added to an oxide "app" that didn't have the appropriate POST sent by the IdP wouldn't be provisioned, and would authenticate correctly but still fail to log in.